server/src/main/java/org/elasticsearch/index/store/FsDirectoryService.java

```diff
@@ -97,11 +92,31 @@ protected Directory newFSDirectory(Path location, LockFactory lockFactory) throw
                 // Use Lucene defaults
                 final FSDirectory primaryDirectory = FSDirectory.open(location, lockFactory);
                 if (primaryDirectory instanceof MMapDirectory) {
-                    return new FileSwitchDirectory(PRIMARY_EXTENSIONS, primaryDirectory, new NIOFSDirectory(location, lockFactory), true) {
+                    return new NIOFSDirectory(location, lockFactory) {
```

Comment:
I think changing the class here requires also to change the assertion in `IndexStoreTests#assertHybridDirectory()`?